### PR TITLE
[WIP] [RFC] [CoreBundle] Resolving channels basing on hostname

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q                    | A
| -------------       | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #3929
| License       | MIT

### CODE REVIEW

File: app/config/config_test.yml [hunk 1]
```diff
@@ -49,3 +49,10 @@ monolog:
 
 twig:
     exception_controller: 'Sylius\Bundle\CoreBundle\Controller\VerboseExceptionController::showAction'
+
+services:
+    sylius.context.channel.test:
+        class: Sylius\Bundle\CoreBundle\Context\TestChannelContext
+        public: false
+        decorates: sylius.context.channel
+        arguments: ["@sylius.context.channel.test.inner"]
```
Comment (pamil) [inline app/config/config_test.yml]: Missing blank line at the end :)

File: features/channels/resolving_channel_by_hostname.feature [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      Given there are following channels configured:
+        | code   | name          | currencies | locales      | url            |
+        | WEB-US | mystore.us    | USD        | en_US        | www.mystore.us |
+        | WEB-EU | mystore.eu    | EUR        | de_DE        | www.mystore.eu |
+        | WEB-FR | mystore.fr    | EUR        | fr_FR        | www.mystore.fr |
+      And I am on the "fr" hostname
+      Then I should be on the "WEB-FR" channel
+
+
+
```
Comment (pamil) [inline features/channels/resolving_channel_by_hostname.feature]: A lot of blank lines at the end :)

File: features/channels/resolving_channel_by_hostname.feature [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            | USD  |
+            | EUR  |
+          And there are following locales configured:
+            | code  |
+            | en_US |
+            | fr_FR |
+            | de_DE |
+          And I am logged in as user
+
+#    Scenario: Trying to view homepage without a default channel configured
```
Comment (lchrusciel) [inline features/channels/resolving_channel_by_hostname.feature]: Why did you commented this scenario?

File: src/Sylius/Component/Channel/UnresolvedChannelException.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class UnresolvedChannelException extends \InvalidArgumentException
+{
+    public function __construct($hostname)
+    {
+        parent::__construct(sprintf('Channel for hostname "%s" cannot be resolved.', $hostname));
+    }
+}
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/UnresolvedChannelException.php]: Missing blank line

File: src/Sylius/Component/Channel/UnresolvedChannelException.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+namespace Sylius\Component\Channel;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class UnresolvedChannelException extends \InvalidArgumentException
+{
+    public function __construct($hostname)
```
Comment (Zales0123) [inline src/Sylius/Component/Channel/UnresolvedChannelException.php]: Missing docblock ;)

**Comment by michalmarcinkowski** [comment]
@TheMadeleine I suppose this can be closed, as we agreed to a little bit different approach :)

**Pull request closed without merging**